samples/sensor/qdec/sample.yaml

```diff
@@ -0,0 +1,15 @@
+sample:
+  description: Usage quadrature decoder sensor
+  name: qdec_sensor
+tests:
+  sample.sensor.qdec_sensor:
+    tags: sensors
+    platform_allow: nucleo_f401re
+    timeout: 10
+    harness: console
```

Comment:
because it requires to connect an external device, harness can't be console per my understanding cc: @PerMac

Reply:
well actually the external device is just a mechanical encoder, so it's not strictly required to initialize the internal timer and run the example. Of course it's mandatory if the user wants to see counter's value changing over time.
Btw, since your point is correct and this feature is hard to auto-test, I changed this part limiting it to build only

Reply:
Harness: console means that twister will look for a given regexes in the console output. If the platform prints a required log in the output, harness console is fine. If an extra hardware is required (e.g. sensor connected, shield or some pin being wired) one needs to define a required `fixture` and twister will only use platforms with has such modification. I saw this is resolved, but since I was pinged I added my 2 cents. https://docs.zephyrproject.org/latest/develop/test/twister.html#fixtures